kernel/kernel-core/src/main/java/io/mosip/kernel/core/cbeffutil/common/CbeffXSDValidator.java

```
@@ -17,15 +22,136 @@
+
+        try {
+            validator.validate(new StreamSource(new ByteArrayInputStream(xmlBytes), "memory:xml"));
+            LOGGER.debug("XML validation successful.");
+            return true;
+        } catch (Exception e) {
+            e.printStackTrace();
+            LOGGER.error("XML validation failed: {}", e.getMessage(), e);
+            throw e;
+        }
+    }
+
+    public static boolean validateXML(final Schema schema, final byte[] xmlBytes) throws Exception {
+        Objects.requireNonNull(schema, "schema");
+        LOGGER.debug("validateXML with precompiled schema={} xmlBytes length={}", schema,
+                (xmlBytes != null ? xmlBytes.length : null));
+
+        requireNonEmpty(xmlBytes, "xmlBytes");
+
+        final Validator validator = getValidator(schema);
+        LOGGER.debug("Validator created: {}", validator);
+
+        try {
+            validator.validate(new StreamSource(new ByteArrayInputStream(xmlBytes), "memory:xml"));
+            LOGGER.debug("XML validation successful with precompiled schema.");
+            return true;
+        } catch (Exception e) {
+            e.printStackTrace();
+            LOGGER.error("Validation failed with precompiled schema: {}", e.getMessage(), e);
+            throw e;
+        }
+    }
+
+    public static Schema compileSchema(final byte[] xsdBytes) throws CbeffException {
+        requireNonEmpty(xsdBytes, "xsdBytes");
+        LOGGER.debug("Compiling schema from xsdBytes length={}", xsdBytes.length);
+        try {
+            Schema schema = SCHEMA_FACTORY.newSchema(new StreamSource(new ByteArrayInputStream(xsdBytes), "memory:xsd"));
+            LOGGER.debug("XSD schema compiled successfully: {}", schema);
+            return schema;
+        } catch (Exception e) {
+            LOGGER.error("Failed to compile XSD schema: {}", e.getMessage(), e);
+            throw new CbeffException("Failed to compile XSD schema::" + e.getLocalizedMessage());

```

Can we throw the Exception as it is to match with existing code and it is not related with CBEFF as well.

---

it is handled correctly